### ISSUE TITLE
Running Over Whole Sets/Computing Epochs Instead of Iterations

### ISSUE THREAD

**Issue by davheld** [BVLC/caffe#1094]
I have an odd number of test images (specifically, I have 672 test images).  If I want to have a batch size of 100, how many test iterations should I choose?  If I pick 6 then we will only iterate through 600 of the 672 test images, but if I pick 7 (to iterate through 700 images) then we will go off the end of the database (though I still get a result, not a segfault).  For the record, it seems that if I pick 7 vs 20 iterations, I get a different result, so it seems that the solver does not naturally just stop once it reaches the end of the test set.  Any help / advice?

**Comment by kuprel** [comment]
One could just use a batch size of 1 for the test set and iterate through all of them, right?  Would that be significantly slower?  I noticed doing this frees up memory, and I'm able to get away with a larger training batch size.  Using gradient accumulation https://github.com/BVLC/caffe/pull/1977 is probably a better way to use a larger training batch size though

**Comment by Coderx7** [comment]
@shelhamer  what happens if someone e.g for 672 images choose test_iter = 7 while having batch size of 100? what would go wrong if someone does that?
